js/index.js

```diff
@@ -38,5 +38,76 @@ const siteContent = {
 };
 
 // Example: Update the img src for the logo
-let logo = document.getElementById("logo-img");
+const logo = document.getElementById("logo-img");
 logo.setAttribute('src', siteContent["nav"]["img-src"])
+
+// update the nav items
+const nav = document.querySelectorAll('nav a');
+nav.forEach((a, i) => {
+  a.textContent = siteContent.nav[`nav-item-${i+1}`];
+  a.style.color = 'green';
+});
+
+// add two more items to the nav
+const firstAnchor = document.createElement('a');
+const lastAnchor = document.createElement('a');
+firstAnchor.textContent = 'First';
+lastAnchor.textContent = 'Last';
+firstAnchor.style.color = 'green';
+lastAnchor.style.color = 'green';
+document.querySelector('header nav').appendChild(lastAnchor);
+document.querySelector('header nav').prepend(firstAnchor);
+
+// updates the cta
+document.querySelector('button').textContent = siteContent.cta["button"];
+document.querySelector('.cta-text h1').innerHTML = 'DOM<br> Is<br> Awesome';
+document.querySelector('#cta-img').src = siteContent.cta["img-src"];
+
+// features
+const features = document.querySelectorAll('.top-content .text-content')[0];
+features.children[0].textContent = siteContent["main-content"]["features-h4"];
+features.children[1].textContent = siteContent["main-content"]["features-content"];
+
+// about
+const about = document.querySelectorAll('.top-content .text-content')[1];
+about.children[0].textContent = siteContent["main-content"]["about-h4"];
+about.children[1].textContent = siteContent["main-content"]["about-content"]
+
+// main content middle image
+document.querySelector('.middle-img').setAttribute('src', siteContent["main-content"]["middle-img-src"]);
+
+// services product and vision
+const services = document.querySelectorAll('.bottom-content .text-content')[0];
+services.children[0].textContent = siteContent["main-content"]["services-h4"];
+services.children[1].textContent = siteContent["main-content"]["services-content"];
+
+const product = document.querySelectorAll('.bottom-content .text-content')[1];
+product.children[0].textContent = siteContent["main-content"]["product-h4"];
+product.children[1].textContent = siteContent["main-content"]["product-content"];
+
+const vision = document.querySelectorAll('.bottom-content .text-content')[2];
+vision.children[0].textContent = siteContent["main-content"]["vision-h4"];
+vision.children[1].textContent = siteContent["main-content"]["vision-content"];
+
+// contact
+document.querySelector('.contact h4').textContent = siteContent["contact"]["contact-h4"];
+document.querySelectorAll('.contact p')[0].textContent = siteContent["contact"]["address"];
+document.querySelectorAll('.contact p')[1].textContent = siteContent["contact"]["phone"];
+document.querySelectorAll('.contact p')[2].textContent = siteContent["contact"]["email"];
```

Comment:
Here would be probably better to select all elements once with querySelectorAll and then calling elements from the array created. 
My personal view on this it's that would be easier to read.